pipelines/matomo/__init__.py

```diff
@@ -0,0 +1,116 @@
+    """
+    The source for the pipeline.
+    :param api_token:
+    :param url:
+    :param live_events_site_id:
+    :param get_live_event_visitors: Option, if set to true will retrieve data about unique visitors.
+    :return: A list of resources containing event data.
+    """
+
+    # Create an instance of the Matomo API client
+    client = MatomoAPIClient(api_token=api_token, url=url)
+    visits_data_generator = get_last_visits(client=client, last_date=dlt.sources.incremental("serverTimestamp"), site_id=live_events_site_id)
+    resource_list = [visits_data_generator]
+    if get_live_event_visitors:
+        resource_list.append(visits_data_generator | get_unique_visitors(client=client, site_id=live_events_site_id))
+    return resource_list
+
+
+@dlt.resource(name="visits", write_disposition="merge", primary_key="idVisit", selected=True)
+def get_last_visits(client: MatomoAPIClient, site_id: int,  last_date: dlt.sources.incremental[pendulum.DateTime], rows_per_page: int = 1000) -> Iterator[TDataItem]:
+    """
+    Dlt resource which gets the visits in the given site id for the given timeframe. If there was a previous load the chosen timeframe is always last_load -> now. Otherwise, a start date can be
+    provided to make the chosen timeframe start_date -> now. The default behaviour would be to get all visits available until now if neither last_load nor start_date are given.
+    :param client: Used to make calls to Matomo API
+    :param site_id: Every site in Matomo has a unique id
+    :param last_date: date of last load for this resource if it exists
+    :returns: Iterator of dicts containing information on last visits in the given timeframe
+    """
+
+    extra_params = {
+        "minTimestamp": last_date.last_value if last_date.last_value else TIMESTAMP_10_DAYS_AGO,
+    }
+    method_data = client.get_method(site_id=site_id, method="Live.getLastVisitsDetails", extra_params=extra_params, rows_per_page=rows_per_page)
+    yield from method_data
+
+
+@dlt.transformer(data_from=get_last_visits, write_disposition="merge", name="visitors", primary_key="visitorId")    # type: ignore
+def get_unique_visitors(visits: List[DictStrAny], client: MatomoAPIClient, site_id: int) -> Iterator[TDataItem]:
+    """
+    Dlt transformer. Receives information about visits from get_last_visits
+    :param visits: List of dicts containing information on last visits in the given timeframe
+    :param client: Used to make calls to Matomo API
+    :param site_id: Every site in Matomo has a unique id
+    :returns: Dict containing information about  the visitor
+    """
+
+    for visit in visits:
+        visitor_id = visit["visitorId"]
+        method_data = client.get_method(site_id=site_id, method="Live.getVisitorProfile", extra_params={"visitorId": visitor_id})
+        yield method_data
```

Comment:
`yield from` because `get_method` yields. otherwise you yield a generator. which `dlt` will deal with by creating additional step in pipeline that we do not need